### PR TITLE
Add support to custom sections on Google docstrings

### PR DESCRIPTION
**Pull request by pslacerda**
It is parsed into a DocstringMeta.

### CODE REVIEW

**Comment by pslacerda** [comment]
In my actual usage I'm monkey patching to support a style similar to google:

```python
import docstring_parser.parser.google, re
docstring_parser.parser.google._sections = {
    "ARGUMENTS": "param",
    "EXAMPLES": "examples",
}
docstring_parser.parser.google._titles_re = re.compile(
    "^(" + "|".join("(%s)" % t for t in docstring_parser.parser.google._sections) + ")", flags=re.M
)
docstring_parser.parser.google._valid = {t for t, a in docstring_parser.parser.google._sections.items() if a}
```

**Comment by rr-** [comment]
@pslacerda can you add tests for this?

**Comment by pslacerda** [comment]
Sure I can. I do need Notes, Note and Example also, beside Examples. Should I add these too?

**Comment by rr-** [comment]
Yes.

If you're feeling adventurous, you could refactor it so that extending the library for whatever sections you need is possible and doesn't require such hacks.

File: docstring_parser/parser/google.py [hunk 1]
```diff
@@ -14,10 +14,10 @@
     "Exceptions": "raises",
     "Except": "raises",
     "Attributes": None,
-    "Example": None,
+    "Example": "examples",
     "Examples": "examples",
     "Returns": "returns",
-    "Yields": "yields",
+    "Yields": "yields"
```
Comment (rr-) [inline docstring_parser/parser/google.py]: Violates [black](https://pypi.org/project/black/)

File: docstring_parser/parser/google.py [hunk 2]
```diff
@@ -115,7 +117,7 @@ def parse(text: str) -> Docstring:
         indent = indent_match.group()
 
         # Check for returns/yeilds (only one element)
-        if _sections[title] in ("returns", "yields"):
+        if _sections[title] in ("returns", "yields", "examples"):
```
Comment (rr-) [inline docstring_parser/parser/google.py]: The comment above is false. Maybe it should be changed to `Check for singular elements` or something?

**Comment by pslacerda** [comment]
I was feeling adventurous, just finished the code. Even with it a bit ugly, it is working for my case. I expect it to be an usual case.

**Comment by pslacerda** [comment]
-- I'd rather preferred GoogleParser.
-- Me too!
[Closed]

File: docstring_parser/parser/google.py [hunk 3]
```diff
@@ -2,27 +2,87 @@
...
+    """For sections like returns or yields."""
+
+
+Section = namedtuple("Section", "title key type")
+
+
+class _globals:
+    sections: T.Dict[str, Section] = None
+    title_colon: bool = None
+    titles_re: re.Pattern = None
```
Comment (rr-) [inline docstring_parser/parser/google.py]: `T.Pattern` cause on Python 3.6.8:

```
../../../.local/lib/python3.6/site-packages/py/_path/local.py:701: in pyimport
    __import__(modname)
docstring_parser/__init__.py:3: in <module>
    from .parser import ParseError, parse
docstring_parser/parser/__init__.py:5: in <module>
    from . import google, rest
docstring_parser/parser/google.py:28: in <module>
    class _globals:
docstring_parser/parser/google.py:31: in _globals
    titles_re: re.Pattern = None
E   AttributeError: module 're' has no attribute 'Pattern'
```

File: docstring_parser/parser/google.py [hunk 4]
```diff
@@ -2,27 +2,87 @@
...
+    valid: T.Set[str] = None
```
Comment (rr-) [inline docstring_parser/parser/google.py]: You initialize these to `None` so the type hints should really be `T.Optional`.

Comment (rr-) [reply docstring_parser/parser/google.py]: Let's refactor it a bit:

```python3
class Config:
    def __init__(self, sections: T.Dict[str, Section], title_colon: bool) -> None:
        self.sections = sections
        self.title_colon = title_colon
        self.titles_re = …
        self.valid = …


config: T.Optional[Config] = None


def setup(sections: T.Optional[T.List[Section]] = None, title_colon: bool = True) -> None:
    global config

    blabla

    config = Config(
        sections=…,
        title_colon=…,
    )


setup()
```

and further later than using `_globals`, use `config`

**Comment by pslacerda** [comment]
I was trying to say that maybe it could be like: 

```python
class GoogleParser:
    """New API for custom sections"""
    _sections: ...
    _title_colon: ...
    _titles_re: ...
    _valid: ...

    def __init__(sections=None, title_colon=True): pass
    def parse(): pass
    def add_section(): pass
    def _build_meta(): pass

def parse(s):
    """Old & easy API keep as it was."""
    return GoogleParser().parse(s)
```

But now I like the way it turned.

File: docstring_parser/parser/google.py (hunk 4 above)
Comment (pslacerda) [reply docstring_parser/parser/google.py]: Refactored with classes because it was becoming weird.

**Comment by pslacerda** [comment]
Just tried with classes, see if you like. Also things that looks like titles but aren't titles are now description.

**Comment by rr-** [comment]
@pslacerda thanks, beautiful work. Merged. LMK if you need a new version on PyPI.

**Comment by pslacerda** [comment]
@rr- seems you merged local but didn't pushed to Github.
I want to show the usage briefly in the README. Then a new version would be more than welcome.

**Comment by rr-** [comment]
@pslacerda haha you're right :D I just merged the changes. OK, LMK once you document the changes and we'll release a new version. Also, I apologize for the late reply.

**Pull request closed without merging**